shell/platform/darwin/ios/ios_surface_metal.mm

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/ios_surface_metal.h"
+#include "flutter/shell/gpu/gpu_surface_metal.h"
+
+namespace flutter {
+
+IOSSurfaceMetal::IOSSurfaceMetal(fml::scoped_nsobject<CAMetalLayer> layer,
+                                 FlutterPlatformViewsController* platform_views_controller)
+    : IOSSurface(platform_views_controller), layer_(std::move(layer)) {}
+
+IOSSurfaceMetal::~IOSSurfaceMetal() = default;
+
+// |IOSSurface|
+bool IOSSurfaceMetal::IsValid() const {
+  return layer_;
+}
+
+// |IOSSurface|
+bool IOSSurfaceMetal::ResourceContextMakeCurrent() {
+  return false;
```

Comment:
Will we be able to have a resource context with metal?

Will we be able to make it the primary one the way we do with OpenGL so that it doesn't get mixed up if we remove/attach ViewControllers?

Reply:
We should be able to support resource contexts. However, unlike OpenGL, there is no need to make it current. The iOS manager can own the context. I haven't verified yet if the Metal backend allows cross context (GrContext) images. If it doesn't, we can still perform image decompressed asynchronously but do the host-to-device transfer on the main context.